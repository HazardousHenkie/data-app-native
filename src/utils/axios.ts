import axios from 'axios'

const Axios = axios.create({
    baseURL: process.env.REACT_APP_API_URL,
})

Axios.defaults.headers.common['Content-Type'] = 'application/json;charset=utf-8'

export default Axios
