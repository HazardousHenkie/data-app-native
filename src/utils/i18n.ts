import i18n from 'i18next'
import * as Localization from 'expo-localization'

import { initReactI18next } from 'react-i18next'

import translationEN from '../../translations/locales/en.json'
import translationJA from '../../translations/locales/ja.json'
import translationNL from '../../translations/locales/nl.json'

const resources = {
    en: translationEN,
    ja: translationJA,
    nl: translationNL,
}

export const appLocales = ['en', 'ja', 'nl']

i18n.use(initReactI18next).init({
    resources,
    debug: true,
    lng: Localization.locale,
    fallbackLng: 'en',
    whitelist: appLocales,
    interpolation: {
        escapeValue: false,
    },
    react: {
        useSuspense: false,
        wait: true,
    },
})

export default i18n
