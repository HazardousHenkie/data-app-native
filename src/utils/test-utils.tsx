import React from 'react'
import { render } from '@testing-library/react-native'

import { Provider } from 'react-redux'

import { ThemeProvider } from 'styled-components'
import { Provider as PaperProvider } from 'react-native-paper'

import { I18nextProvider } from 'react-i18next'

import i18n from './i18nTesting'

import darkTheme from '../styles/themeStyles'
import variables from '../styles/variables'
import configureStore from '../store/configureStore'

const customRender = (
    ui: React.ReactElement,
    {
        initialState = {},
        store = configureStore(initialState),
        ...renderOptions
    } = {}
) => {
    const Wrapper: React.FC = ({ children }) => {
        return (
            <Provider store={store}>
                <I18nextProvider i18n={i18n}>
                    <ThemeProvider theme={variables}>
                        <PaperProvider theme={darkTheme}>
                            {children}
                        </PaperProvider>
                    </ThemeProvider>
                </I18nextProvider>
            </Provider>
        )
    }

    return render(ui, {
        wrapper: Wrapper,
        ...renderOptions,
    })
}

export * from '@testing-library/react-native'

export { customRender as render }
