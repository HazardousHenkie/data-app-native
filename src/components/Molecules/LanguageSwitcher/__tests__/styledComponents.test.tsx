import React from 'react'

import LanguageSwitcherWrapper from '../styledComponents'

import 'jest-styled-components'
import { render } from '../../../../utils/test-utils'

describe('<LanguageSwitcherWrapper />', () => {
    it('should render and match the snapshot', () => {
        const { toJSON } = render(<LanguageSwitcherWrapper />)

        expect(toJSON).toMatchSnapshot()
    })
})
