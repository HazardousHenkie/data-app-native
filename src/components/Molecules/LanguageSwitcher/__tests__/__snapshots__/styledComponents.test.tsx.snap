// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageSwitcherWrapper /> should render and match the snapshot 1`] = `[Function]`;
